### PR TITLE
[Server] Models: Component Generator Optimizations

### PR DESCRIPTION
**Pull request by AlexanderSanin**
1 Simplified Conditional Checks: Combined conditions to reduce redundancy.
2 Optimized String Building: Used strings.Builder more effectively to build messages.
3 Error Handling: Simplified error message construction by directly appending to the slice.
4 Switch Statement: Used a switch statement in determinePluralWord for better readability.

**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by AlexanderSanin** [comment]
@aabidsofi19 please review and add "hachtoberfest-accepted" label

**Review by leecalcote — COMMENTED** [review]
Thank you, @AlexanderSanin!

File: server/handlers/componetgenratorhelper.go [hunk 1]
```diff
@@ -425,17 +415,21 @@ func ErrMsgContruct(response *models.RegistryAPIResponse) string {
 	msg += ")"
 	return msg
 }
+
 func determinePluralWord(count int, word string) string {
```
Comment (MUzairS15) [inline server/handlers/componetgenratorhelper.go:419]: Such logic should reside in `meshkit` alongside other helper functions like formatToCamelCase....

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

**Comment by vishalvivekm** [comment]
@AlexanderSanin 
Thank you for your contributions! Let's discuss this during the Meshery Development call today at 9:00 AM CT. Adding it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?pli=1#) || [meeting link](https://meet.layer5.io/meshery).

**Review by aabidsofi19 — APPROVED** [review]

File: server/handlers/componetgenratorhelper.go [hunk 2]
```diff
@@ -356,67 +356,57 @@ func (h *Handler) sendEventForImport(userID uuid.UUID, provider models.Provider,
...
 		message.WriteString("(")
-	}
-	if response.EntityCount.CompCount > 0 {
-		componentWord := determinePluralWord(response.EntityCount.CompCount, "component")
-		message.WriteString(fmt.Sprintf("%d %s", response.EntityCount.CompCount, componentWord))
-	}
-	if response.EntityCount.RelCount > 0 && response.EntityCount.CompCount > 0 {
-		if message.Len() > 0 {
-			message.WriteString(" and ")
+		if response.EntityCount.CompCount > 0 {
```
Comment (aabidsofi19) [inline server/handlers/componetgenratorhelper.go:371]: we could use templating to clean up this  string manipulation logic.

File: server/handlers/componetgenratorhelper.go [hunk 3]
```diff
@@ -356,67 +356,57 @@ func (h *Handler) sendEventForImport(userID uuid.UUID, provider models.Provider,
...
-	componentsPresent := response.EntityCount.ErrCompCount > 0
-	relationshipsPresent := response.EntityCount.ErrRelCount > 0
-	modelsPresent := response.EntityCount.ErrModelCount > 0
-	knownErrors := response.EntityCount.ErrCompCount + response.EntityCount.ErrRelCount + response.EntityCount.ErrModelCount
-	unknownErrors := response.EntityCount.TotalErrCount - knownErrors
 
-	// Collect errors in a slice for dynamic message construction
 	errors := []string{}
-	if modelsPresent {
+	if response.EntityCount.ErrModelCount > 0 {
```
Comment (aabidsofi19) [inline server/handlers/componetgenratorhelper.go:393]: i think keeping these as variables is better for readabilty and debugging . there is no need for this .

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

**Comment by AlexanderSanin** [comment]
@aabidsofi19 sorry, but I wasn't able to join the call and rigjt now very busy, so can't contribute more to this project. You can close my pr, thank you

**Pull request closed without merging**